### DISCUSSION TITLE
Error for Theme for book

### DISCUSSION THREAD

**Discussion by samyarmodabber** [quarto-dev/quarto-cli#4296] [Q&A]
Hello friends,
I use CLI and create a book. But when I want to see a preview it gives me an error:

$ quarto preview
Preparing to preview
[1/4] index.qmd
ERROR: Theme file compilation failed:

Could you help me?
thanks

**Comment by dragonstyle** [comment]
This is a likely an issue related to SCSS themes - if you can share the source code, it will help a lot.

**Reply by samyarmodabber** [reply]
In fact, I did not do a lot.
1. I install Quarto CLI on my windows
2. I add Quarto extension on VSC
3. and did steps of document: https://quarto.org/docs/books

I put the code here:
https://github.com/samyarmodabber/FM

**Reply by cscheid** [reply]
Can you run `quarto check` on your command line and paste the result?

**Reply by samyarmodabber** [reply]
quarto check

[>] Checking Quarto installation......OK
      Version: 1.2.335
      Path: C:\Users\Sam Modabber\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking basic markdown render....OK

ERROR: Theme file compilation failed:

**Reply by cscheid** [reply]
This feels like a quarto bug of some sort. Can you type `set QUARTO_PRINT_STACK=true` (regular cmd prompt) or `$env:QUARTO_PRINT_STACK = "true"` (powershell) and then run that command again?

The only thing I can see that is suspicious is the fact that your username has a space in it.

**Reply by samyarmodabber** [reply]
[>] Checking Quarto installation......OK
      Version: 1.2.335
      Path: C:\Users\Sam Modabber\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking basic markdown render....OK

ERROR: Theme file compilation failed:



Error: Theme file compilation failed:


    at dartCommand (file:///C:/Users/Sam%20Modabber/AppData/Local/Programs/Quarto/bin/quarto.js:62720:15)
    at async dartCompile (file:///C:/Users/Sam%20Modabber/AppData/Local/Programs/Quarto/bin/quarto.js:62686:5)
    at async compileWithCache (file:///C:/Users/Sam%20Modabber/AppData/Local/Programs/Quarto/bin/quarto.js:62906:17)
    at async compileSass (file:///C:/Users/Sam%20Modabber/AppData/Local/Programs/Quarto/bin/quarto.js:62772:12)
    at async resolveSassBundles (file:///C:/Users/Sam%20Modabber/AppData/Local/Programs/Quarto/bin/quarto.js:68040:27)
    at async resolveExtras (file:///C:/Users/Sam%20Modabber/AppData/Local/Programs/Quarto/bin/quarto.js:69427:18)
    at async runPandoc (file:///C:/Users/Sam%20Modabber/AppData/Local/Programs/Quarto/bin/quarto.js:69053:24)
    at async renderPandoc (file:///C:/Users/Sam%20Modabber/AppData/Local/Programs/Quarto/bin/quarto.js:69648:26)
    at async Object.onRender (file:///C:/Users/Sam%20Modabber/AppData/Local/Programs/Quarto/bin/quarto.js:79336:32)
    at async renderFiles (file:///C:/Users/Sam%20Modabber/AppData/Local/Programs/Quarto/bin/quarto.js:78709:21)

**Reply by dragonstyle** [reply]
I don't have much of an explanation for what is going on - it looks like this we're trying compile an SCSS file which is for some reason throwing an error. The big mystery to me is the lack of a better exception message.

Can you manually run our sass compiler from the terminal/powershell? It is at `C:\Users\Sam Modabber\AppData\Local\Programs\Quarto\bin\tools\dart-sass\sass.bat`?

**Reply by cderv** [reply]
I wonder if the special char in username (a space) can be the culprit here. Often, such space or other character can cause issue, especially if somewhere we are not quoting correctly 🤔

**Reply by dragonstyle** [reply]
I was suspicious and tried placing Quarto itself in a path with a space but didn't run into any issues. I didn't try a temp path with spaces in it, however, so I suppose that is possible...

**Reply by cderv** [reply]
@dragonstyle do we have a way to have more log from dart itself ?  I can't reproduce either

**Reply by cderv** [reply]
Also @samyarmodabber just to understand correctly: You get the issue also when just running `quarto check` ?

**Reply by dragonstyle** [reply]
> @dragonstyle do we have a way to have more log from dart itself ? I can't reproduce either

Right not really - we don't forward additional verbosity options through to it, though I think that is a very good idea. Typically the underlying error includes additional information (e.g. a missing file or invalid syntax or something) so I'm confused why the error message is so bad.

**Reply by cderv** [reply]
We had the issue in the past and error was better : example in  https://github.com/quarto-dev/quarto-cli/issues/338 and more in [search](https://github.com/search?q=repo%3Aquarto-dev%2Fquarto-cli+ERROR%3A+Theme+file+compilation+failed%3A&type=issues)

Not sure why so minimal now

**Reply by dragonstyle** [reply]
Yes - I feel like it may be related to the cause of the error. In my simple windows tests, when I introduce a syntax error, etc... we are still printing a complete error message. I'm guessing that somehow dart itself is not returning a complete error message...

**Reply by cderv** [reply]
Ok. We probably need more information from what triggers the issue @samyarmodabber. Thanks

**Reply by samyarmodabber** [reply]
> Also @samyarmodabber just to understand correctly: You get the issue also when just running `quarto check` ?

yes,  I get the same error:
ERROR: Theme file compilation failed:

**Reply by samyarmodabber** [reply]
> I wonder if the special char in username (a space) can be the culprit here. Often, such space or other character can cause issue, especially if somewhere we are not quoting correctly 🤔

Yes. It is my problem. I made a new account without space and it works. 

But after that, I try to change my username from "Sam Modabber" to "SamModabber" and after changing again not work.

I think I have to keep the new windows account and run QUARTO from that. 

Thank you for your help

**Reply by cderv** [reply]
> Yes. It is my problem. I made a new account without space and it works.

Ok thanks for confirmation

> I think I have to keep the new windows account and run QUARTO from that.

Seems like a good workaround for you, but probably not that convenient. We'll try to go to the bottom of this if we manage to reproduce.
